gui/scripts/syncthing/core/services/localeService.js

```diff
@@ -18,6 +18,20 @@ angular.module('syncthing.core')
             _availableLocales = locales;
         };
 
+        function isLocalStorageEnabled () {
+            // Feature detect localStorage; https://mathiasbynens.be/notes/localstorage-pattern
+            try {
+                var uid = new Date();
+                var storage = window.localStorage;
+                storage.setItem(uid, uid);
+                var success = storage.getItem(uid) == uid;
+                storage.removeItem(uid);
+                return success;
+            } catch (exception) {
+                return false;
+            }
+        }
+
```

Comment:
How about returning `storage` instead, and assigning it to for example `_storage` in the provider scope (although I'm not sure why we have underscored names here...)? Then just `if (_storage) { // usage _storage }` and we don't need to rerun the test on each usage.